herd/AArch64Sem.ml

```diff
@@ -2399,6 +2399,25 @@ module Make
                 (write_reg r1 v2 ii)
                 (csel_op op v3 >>= mask (fun v ->  write_reg r1 v ii)))
 
+        | I_UBFM (v,rd,rn,kr,ks) ->
+          let sz = tr_variant v in
+          (* Make bitmask, print as hex, apply AndK to zero out upper/lower bits *)
+          let bitmask = match v with
+           | V32 -> List.init 32 (fun v -> if v <= kr || v >= ks then "0" else "1")
+           | V64 -> List.init 64 (fun v -> if v <= kr || v >= ks then "0" else "1")
+           | _ -> assert false in
+          let bitmask = List.rev bitmask in
+          let dec_mask = int_of_string (Printf.sprintf "0b%s" (String.concat "" bitmask)) in
+          let hex_mask = Printf.sprintf "0x%x" dec_mask in
+          let rec log2 x = match x with
+            | 1 -> 0
+            | _ -> 1 + log2 ((x+1)/2) in
+          read_reg_data sz rn ii
+          >>= fun v -> M.op1 (Op.AndK hex_mask) v
+          >>= fun v -> M.op Op.ShiftRight (V.intToV (log2 kr-ks)) v
+          >>= fun v -> write_reg rd v ii
+          >>= B.next1T
```

Comment:
I do not have time for a complete review. Here is a remark. Computation at the assembler level are in `Int64`. Hence, one can use the same technique as above, but on integers of 64 bit size. For instance `int_of_string` can be replaced by `Int64.of_string` and format `"0x%x"` by `"0x%Lx"`. However I have not checked boundary cases.

Besides, I am not so sure about using the `log2` function. Shouldn't the shift amount be `kr-ks`?

Finally shouldn't there be two cases depending upon the relative position of kr and ks?